### DISCUSSION TITLE
Sensor count of consecutive power-on power-off cycles

### DISCUSSION THREAD

**Discussion by jaymemaurice** [arendst/Tasmota#11232] [Ideas]
Hey guys,
I want to put wifi smart bulbs in my garage door openers as a quick way to trigger automations.
The low hanging fruit is turning on garage/mud room lights when opener lights are on, or off when opener lights turn off so there is no need to turn off the mudroom before exiting the garage

I know we trap on the power-on-off-on-off-on-off-on for safe mode... and we can change the count of power cycles needed.
What I would like is an exposed sensor that can show the number of previous consecutive power-on-off-on events...
That way I could know if someone tripped the garage door crush/entrapment sensors while it was closing as it's signaled through light flashes.

Before I dig, I was wondering if anyone knows if such functionality already exists?

**Comment by sfromis** [comment]
No such feature in Tasmota. If you're not into much tinkering, what could work would be the Tasmota-compatible [Shelly i3](https://templates.blakadder.com/shelly_i3.html) able to react to pulsing input.

**Reply by jaymemaurice** [reply]
Nah man, I'm into tinkering. This can be done in software though.
I want to screw a bulb into the bulb slot, not wire up stuff into the garage door openers. If I were going to wire things into the garage door opener, I'd probably put a $2 ESP12s directly into it's low voltage brain... not the high voltage side.
The beauty of just using a smart bulb is that everything is still UL/Intertek certified and I didn't create any burn the house down liability.
The garage door opener I am talking about is one of the trolley style american garage door openers which is typically not professionally installed as it relies on the garage door being properly counter balanced... it just pulls the door along and has a light and sensors to make sure the entrapment zone is clear. It has a bulb socket inside which turns on when the garage door is opened, the light button is pushed on the opener switch or remote, or the entrapment zone's sensors are interrupted.
![image](https://user-images.githubusercontent.com/38895319/110135711-d4977d00-7d9c-11eb-93f4-9229ede005a9.png)

**Reply by sfromis** [reply]
Dig [here](https://github.com/arendst/Tasmota/blob/development/tasmota/settings.ino#L305-L335) :-)

**Reply by ascillato** [reply]
> ... UL/Intertek certified and I didn't create any burn the house down liability...

Be aware that when using alternative firmwares, the UL certification expires. The certification is tied ONLY to the original firmware. Obviously, if the bulb and everything burns, there is no way to know that you have modified the bulb's firmware but depending on the insurance company, they could be very picky on that. Just a FYI.

**Reply by ascillato** [reply]
I agree with @sfromis. Using a shelly (that is UL certified) will be the easiest way to accomplish what you want, and will be faster and more reliable than the On Off counting.

**Reply by jaymemaurice** [reply]
@ascillato I guess my stumbling point is the wiring... although the shelly may be ul certified, there is no direct way to put it in the light socket short of opening up an modifying the opener or using a e17 adaptor which creates fitment issues with the light that I still want in the fixture. Further the shelley would need to be powered on continuously so I would need not only to tap into the power for the light which the opener switches on and off, but also a constant power source. 

Modifying the firmware of the light vs modifying the burn down house wiring and kid crushing garage door opener in a way that it was not intended... vs. doing nothing.

Another way to achieve my goal might be one of those power monitoring sockets... just monitor the power draw to know if the lights turn on... or garage door is moving...

But anyway counting power-on-off cycles also would allow other cool useful automations like making a dumb lamp work like one of those old 3 position dimmer incandescent toggling it's brightness based on the number of times you turned it on.

I'll give it a go if I get some free time and submit a PR if I make it work well. Is there anything you can think of that you would want to see to make it acceptable (any helpful test requirements)? or anything you definitely don't want to see?

**Reply by sfromis** [reply]
Yet another strategy, without any invasion into the wiring, could be a simple dev board with a light sensor.
